samcli/lib/samlib/resource_metadata_normalizer.py

```diff
@@ -56,10 +72,59 @@ def _replace_property(property_key, property_value, resource, logical_id):
 
         """
         if property_key and property_value:
-            resource.get(PROPERTIES_KEY, {})[property_key] = property_value
+            nested_keys = property_key.split(".")
+            target_dict = resource.get(PROPERTIES_KEY, {})
+            while len(nested_keys) > 1:
+                key = nested_keys.pop(0)
+                target_dict[key] = {}
+                target_dict = target_dict[key]
+            target_dict[nested_keys[0]] = property_value
         elif property_key or property_value:
             LOG.info(
                 "WARNING: Ignoring Metadata for Resource %s. Metadata contains only aws:asset:path or "
                 "aws:assert:property but not both",
                 logical_id,
             )
+
+    @staticmethod
+    def _extract_image_asset_metadata(metadata):
+        """
+        Extract/create relevant metadata properties for image assets
+
+        Parameters
+        ----------
+        metadata dict
+            Metadata to use for extracting image assets properties
+
+        Returns
+        -------
+        dict
+            metadata properties for image-type lambda function
+
+        """
+        path_from_asset, dockerfile = os.path.split(metadata.get(ASSET_DOCKERFILE_PATH_KEY))
+        asset_path = metadata.get(ASSET_PATH_METADATA_KEY)
+        dockerfile_context = os.path.join(asset_path, path_from_asset)
```

Comment:
What happens in asset_path is `None`?

Reply:
In theory it shouldn't ever be None, but if it is then the Lambda function image won't get built, and it'll fail when trying to local invoke.